### PR TITLE
refactor: begin tearing apart AutopushSettings

### PR DESCRIPTION
**Pull request by pjenvey**
- move db concerns (incl. metrics) into a simple DatabaseManager.
heavier db init's moved into its own setup() method that most tests
don't even call
- split out routers, now connection doesn't even create them
- kill PushState.msg_limit, nor did it need a metrics attrib

issue #632

### CODE REVIEW

File: autopush/router/__init__.py [hunk 1]
```diff
@@ -12,3 +12,18 @@
 
 __all__ = ["APNSRouter", "FCMRouter", "GCMRouter", "SimpleRouter",
            "WebPushRouter"]
+
+
+def routers_from_settings(settings, db):
+    """Create a dict of IRouters for the given settings"""
+    router_conf = settings.router_conf
+    routers = dict(
+        simplepush=SimpleRouter(
```
Comment (pjenvey) [inline autopush/router/__init__.py]: Sure but not right now

File: autopush/db.py [hunk 2]
```diff
@@ -853,3 +861,130 @@ def clear_node(self, item):
...
+        """Initializes a dict of the initial rotating messages tables.
+
+        An entry for last months table, an entry for this months table,
+        an entry for tomorrow, if tomorrow is a new month.
+
+        """
+        today = datetime.date.today()
+        last_month = get_rotating_message_table(self._message_prefix, -1)
+        this_month = get_rotating_message_table(self._message_prefix)
+        self.current_month = today.month
```
Comment (pjenvey) [inline autopush/db.py]: exactly what I'm trying to get at -- I actually did forget that Router/Storage are still being initialized upon creation here. I'll fix this later in another commit. DatabaseManager() creation should be light weight (no network/db calls), until you explicitly tell it to initialize

File: autopush/db.py [hunk 3]
```diff
@@ -853,3 +861,130 @@ def clear_node(self, item):
...
+
+    @message.setter
+    def message(self, value):
+        """Setter to set the current message table"""
+        self.message_tables[self.current_msg_month] = value
+
+    def _tomorrow(self):
+        return datetime.date.today() + datetime.timedelta(days=1)
+
+    def create_initial_message_tables(self):
```
Comment (pjenvey) [inline autopush/db.py]: forgot some of these and we do have some circular imports. this is going to start getting typing.TYPE_CHECKING ugly

Comment (bbangert) [inline autopush/db.py]: Ah, ok.

File: autopush/db.py (hunk 2 above)
Comment (bbangert) [inline autopush/db.py]: Sounds good.

File: autopush/base.py [hunk 4]
```diff
@@ -1,21 +1,41 @@
 import sys
 import uuid
+from typing import TYPE_CHECKING
 
 import cyclone.web
 from twisted.logger import Logger
 from twisted.python import failure
 
+if TYPE_CHECKING:  # pragma: nocover
```
Comment (pjenvey) [inline autopush/base.py]: sweet

File: autopush/db.py [hunk 5]
```diff
@@ -853,3 +861,132 @@ def clear_node(self, item):
...
+    metrics = attrib()  # type: IMetrics
+
+    message_tables = attrib(default=Factory(dict))  # type: Dict[str, Message]
+    current_msg_month = attrib(default=None)        # type: Optional[str]
+    current_month = attrib(default=None)            # type: Optional[int]
+
+    _message_prefix = attrib(default="message")  # type: str
+
+    @classmethod
+    def from_settings(cls, settings):
```
Comment (pjenvey) [inline autopush/db.py]: we get a light weight `__init__` automatically w/ @attrs (some tests use it). these might change around a bit though in a later commit, re the deferred initialization comments above

Comment (jrconlin) [inline autopush/db.py]: Ah! ok, kinda like how we have to have `ap_settings`.

**Pull request merged**